vllm/model_executor/models/llama.py

```diff
@@ -257,17 +257,23 @@ def __init__(
                       (lora_config.max_loras or 1)) if lora_config else 0
         self.vocab_size = config.vocab_size + lora_vocab
         self.org_vocab_size = config.vocab_size
-        self.embed_tokens = VocabParallelEmbedding(
-            self.vocab_size,
-            config.hidden_size,
-            org_num_embeddings=config.vocab_size,
-        )
+        if get_pp_group().is_first_rank:
```

Comment:
`if get_pp_group().is_first_rank or (config.tie_word_embeddings and get_pp_group().is_last_rank)` ? we need to take care of tie word embedding as well.